src/Elastic.Apm.AspNetCore/DiagnosticListener/AspNetCoreDiagnosticListener.cs

```diff
@@ -11,8 +11,7 @@ public class AspNetCoreDiagnosticListener : IDiagnosticListener
 	{
 		private readonly AbstractLogger _logger;
 
-		public AspNetCoreDiagnosticListener()
-			=> _logger = Agent.CreateLogger(Name);
+		public AspNetCoreDiagnosticListener(AbstractLogger logger) => _logger = logger;
```

Comment:
Similarly to ElasticCoreListeners.cs I think we should make this optional and default to Agent.Config.Logger (or Agent.Config if we change it...).